### PR TITLE
Add approval tests

### PR DESCRIPTION
**Pull request by agluszak**
Approval tests ensure that we don't accidentally change the generated spec.

### CODE REVIEW

**Comment by abrams27** [comment]
soo they will fail for breaking changes / features, right?

**Comment by agluszak** [comment]
They fail if the generated spec is changed. A change must be approved by changing the *expected* output. See https://approvaltests.com/

**Review by ckipp01 — APPROVED** [review]
Interesting, I haven't seen this before. Looks good 👍🏼

**Comment by ckipp01** [comment]
Might also be worth adding a note about the new tests in the [Contributing docs](https://github.com/build-server-protocol/build-server-protocol/blob/master/CONTRIBUTING.md).

**Comment by agluszak** [comment]
> Might also be worth adding a note about the new tests in the [Contributing docs](https://github.com/build-server-protocol/build-server-protocol/blob/master/CONTRIBUTING.md).

Done

File: codegen/src/test/scala/DocsApprovalTest.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    // Get the list of all filenames (without file extensions) in the
+    // spec/src/main/resources/META-INF/smithy/bsp/extensions directory.
+    val path = Paths.get("spec/src/main/resources/META-INF/smithy/bsp/extensions")
+    val filenames = Files
+      .list(path)
+      .toScala(List)
+      .map(_.getFileName.toString.split("\\.").head)
+
+    if (filenames.isEmpty) {
+      throw new RuntimeException("No extensions found, make sre the path is correct.")
```
Comment (jastice) [inline codegen/src/test/scala/DocsApprovalTest.scala:43]: typo: make sre

**Review by jastice — COMMENTED** [review]
thanks! I do remember that such tests had a way too high maintenance load in the sbt-structure project, but probably the problems there don't apply to these tests

**Pull request merged**